server/ntb/publish/ntb_ninjs.py

```diff
@@ -23,46 +17,94 @@ def __init__(self):
+
+        if ninjs.get("description_text"):
+            ninjs["descriptions"] = self.format_descriptions(ninjs)
+
+        if ninjs.get("body_html"):
+            ninjs["bodies"] = self.format_bodies(ninjs)
+
+        if ninjs.get("subject"):
+            ninjs["subjects"] = self.format_subjects(ninjs)
+
+        # removable items which mapped according to Ninjs v2
+        removable_items = [
+            "body_html",
+            "description_html",
+            "description_text",
+            "charcount",
+            "wordcount",
+            "readtime",
+            "headline",
+            "subject",
+        ]
+        for item in removable_items:
+            ninjs.pop(item)
 
         return ninjs
 
     def format_people(self, article):
         peoples = []
-        persons = article["person"]
-        for item in persons:
-            peoples.append(
-                {
-                    "name": item.get("name"),
-                    "rel": item.get("description"),
-                    "qcode": item.get("qcode"),
-                    "source": item.get("source"),
-                }
-            )
-        return peoples
-
-    def format_organisation(self, article):
+        if article.get("person"):
+            persons = article["person"]
+            for item in persons:
+                altids = item.get("altids")
+                peoples.append(
+                    {
+                        "name": item.get("name"),
+                        "rel": item.get("description"),
+                        "qcode": item.get("qcode")
+                        if altids.get("wikidata") is None
+                        else altids["wikidata"],
+                        "source": item.get("source"),
+                    }
+                )
+            return peoples
+
+    def format_organisations(self, article):
         organisations = []
-        items = article["organisation"]
-        for item in items:
-            organisations.append(
-                {
-                    "name": item.get("name"),
-                    "rel": item.get("description"),
-                    "qcode": item.get("qcode"),
-                    "source": item.get("source"),
-                }
-            )
-        return organisations
+        if article.get("organisation"):
+            items = article["organisation"]
+            for item in items:
+                altids = item.get("altids")
+                organisations.append(
+                    {
+                        "name": item.get("name"),
+                        "rel": item.get("description"),
+                        "qcode": item.get("qcode")
+                        if altids.get("wikidata") is None
+                        else altids["wikidata"],
+                        "source": item.get("source"),
+                    }
+                )
+            return organisations
+
+    def format_headlines(self, article):
+        return [{"value": article.get("headline")}]
```

Comment:
we can also set the contenttype here

Reply:
Based on the NinJS v2.0 standard, the key is 'contenttype' and the value is an IANA MIME type (i.e. `text/plain`), same goes for the `format_descriptions` function